lib/ansible/modules/cloud/azure/azure_rm_cdnendpoint.py

```diff
@@ -0,0 +1,677 @@
+
+        resource_group = self.get_resource_group(self.resource_group)
+        if not self.location:
+            self.location = resource_group.location
+
+        if self.query_string_caching_behavior:
+            self.query_string_caching_behavior = _snake_to_camel(self.query_string_caching_behavior)
+
+        response = self.get_cdnendpoint()
+
+        if self.state == 'present':
+
+            if not response:
+
+                if self.started is None:
+                    # If endpoint dosen't exist and no start/stop operation specified, create endpoint.
+                    if self.origins is None:
+                        self.fail("Origins is not provided when trying to create endpoint")
+                    self.log("Need to create the Azure CDN endpoint")
+
+                    if not self.check_mode:
+                        result = self.create_cdnendpoint()
+                        self.results['id'] = result['id']
+                        self.results['host_name'] = result['host_name']
+                        self.log("Creation done")
+
+                    self.results['changed'] = True
+                    return self.results
+
+                else:
+                    # Fail the module when user try to start/stop a non-existed endpoint
+                    self.log("Can't stop/stop a non-existed endpoint")
+                    self.fail("This endpoint is not found, stop/start is forbidden")
+
+            else:
+                self.log('Results : {0}'.format(response))
+                update_tags, response['tags'] = self.update_tags(response['tags'])
+
+                if response['provisioning_state'] == "Succeeded":
+
+                    if self.started is False and response['resource_state'] == 'Running':
+                        self.log("Need to stop the Azure CDN endpoint")
+
+                        if not self.check_mode:
+                            result = self.stop_cdnendpoint()
+                            self.results['id'] = result['id']
+                            self.results['host_name'] = result['host_name']
+                            self.log("Endpoint stopped")
+
+                        self.results['changed'] = True
+                        return self.results
+
+                    elif self.started and response['resource_state'] == 'Stopped':
+                        self.log("Need to start the Azure CDN endpoint")
+
+                        if not self.check_mode:
+                            result = self.start_cdnendpoint()
+                            self.results['id'] = result['id']
+                            self.results['host_name'] = result['host_name']
+                            self.log("Endpoint started")
+
+                        self.results['changed'] = True
+                        return self.results
+
+                    elif self.started is not None:
+                        self.log("Start/Stop not performed due to current resource state")
```

Comment:
Should this be a warning? The resource is not in a manageable (or possibly usable) state...

Reply:
fixed